lib/fastlane/plugin/wpmreleasetoolkit/versioning/files/android_version_file.rb

```diff
@@ -0,0 +1,88 @@
+        attr_reader :version_properties_path
+
+        # Initializes a new instance of AndroidVersionFile with the specified version.properties file path.
+        #
+        # @param [String] version_properties_path The path to the version.properties file.
+        #
+        def initialize(version_properties_path:)
+          @version_properties_path = version_properties_path
+        end
+
+        # Reads the version name from a version.properties file.
+        #
+        # @return [String] The version name read from the file.
+        #
+        # @raise [UI::Error] If the file_path is nil or the version name is not found.
+        #
+        def read_version_name
+          verify_version_properties_exists
+
+          # Read the version name from the version.properties file
+          file_content = JavaProperties.load(version_properties_path)
+          version_name = file_content[:versionName]
+          UI.user_error!('Version name not found in version.properties') if version_name.nil?
+
+          version_name
+        end
+
+        # Reads the version code from a version.properties file.
+        #
+        # @param file_path [String] The path to the version.properties file.
+        #
+        # @return [BuildCode] An instance of `BuildCode` representing the version code read from the file.
+        #
+        # @raise [UI::Error] If the file_path is nil or the version code is not found.
+        #
+        def read_version_code
+          verify_version_properties_exists
+
+          # Read the version code from the version.properties file
+          file_content = JavaProperties.load(version_properties_path)
+          version_code = file_content[:versionCode]
+          UI.user_error!('Version code not found in version.properties') if version_code.nil?
+
+          # Create a BuildCode object
+          Fastlane::Models::BuildCode.new(version_code)
+        end
+
+        # Writes the provided version name and version code to the version.properties file.
+        #
+        # @param version_name [String] The version name to write to the file.
+        # @param version_code [String] The version code to write to the file.
+        #
+        # @raise [UI::Error] If the version name or version code is nil.
+        #
+        def write_version(version_name, version_code)
+          verify_version_properties_exists
+
+          # Create the version name and version code hash
+          version = {
+            versionName: version_name,
+            versionCode: version_code
+          }
+
+          # Write the version name and version code hash to the version.properties file
+          JavaProperties.write(
+            version,
+            version_properties_path
+          )
+        end
+
+        # Verifies the existence of the version.properties file.
+        #
+        # @raise [UI.user_error] Raised if the version.properties file does not exist.
+        #
+        def verify_version_properties_exists
```

Comment:
I ended up moving the `.exists?` check into the initializer because these methods in these classes always take in the version file. It seemed to make more sense to do the validation there instead of having these calls throughout the class.

https://github.com/wordpress-mobile/release-toolkit/commit/5f77bed1c17e2a79ea50a3a9c0de48f4b7b5d8e9